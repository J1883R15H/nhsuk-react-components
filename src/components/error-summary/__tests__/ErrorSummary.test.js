import React from 'react';
import { shallow, mount } from 'enzyme';
import sinon from 'sinon';
import ErrorSummary from '../error-summary';

describe('ErrorSummary', () => {
  it('item matches snapshot', () => {
    const shallowMount = shallow(
      <ErrorSummary.Item href="#">Error item</ErrorSummary.Item>
    );
    expect(shallowMount).toMatchSnapshot();
    shallowMount.unmount();
  });
  it('handles onClick properly', () => {
    const onClick = sinon.spy();
    const shallowMount = shallow(
      <ErrorSummary.Item onClick={onClick}>Error item</ErrorSummary.Item>
    );
    shallowMount.find('a').simulate('click');
    sinon.assert.called(onClick);
    shallowMount.unmount();
  });
  it('all matches snapshot', () => {
    const mountedElement = mount(
      <ErrorSummary
        title="This is a problem"
        description="Optional description of the errors and how to correct them"
      >
        <ErrorSummary.Item href="#example-error-1">
          Link to error with explanation
        </ErrorSummary.Item>
        <ErrorSummary.Item href="#example-error-2">
          Link to error with explanation
        </ErrorSummary.Item>
      </ErrorSummary>
    );
    expect(mountedElement).toMatchSnapshot();
    mountedElement.unmount();
  });
  it("doesn't render description or title when not provided", () => {
    const mountedElement = mount(
      <ErrorSummary>
        <ErrorSummary.Item href="#example-error-1">
          Link to error with explanation
        </ErrorSummary.Item>
        <ErrorSummary.Item href="#example-error-2">
          Link to error with explanation
        </ErrorSummary.Item>
      </ErrorSummary>
    );
    expect(mountedElement).toMatchSnapshot();
    mountedElement.unmount();
  });
  it('responds to onClick', () => {
    const clickFn = jest.fn();
    const wrapper = mount(
      <ErrorSummary>
        <ErrorSummary.Item onClick={clickFn}>Item</ErrorSummary.Item>
      </ErrorSummary>
    );
    const wrapperTest = mount(
      <ErrorSummary>
        <ErrorSummary.Item>Item</ErrorSummary.Item>
      </ErrorSummary>
    );
    wrapper.find('a').simulate('click');
    wrapperTest.find('a').simulate('click');
    expect(clickFn).toHaveBeenCalled();
    wrapper.unmount();
    wrapperTest.unmount();
  });
});
