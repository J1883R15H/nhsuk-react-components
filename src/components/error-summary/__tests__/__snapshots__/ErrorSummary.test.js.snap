// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorSummary all matches snapshot 1`] = `
<ErrorSummary
  className=""
  description="Optional description of the errors and how to correct them"
  role="alert"
  tabIndex="-1"
  title="This is a problem"
>
  <div
    className="nhsuk-error-summary"
    role="alert"
    tabIndex="-1"
  >
    <h2
      className="nhsuk-error-summary__title"
    >
      This is a problem
    </h2>
    <div
      className="nhsuk-error-summary__body"
    >
      <p>
        Optional description of the errors and how to correct them
      </p>
      <ul
        className="nhsuk-list nhsuk-error-summary__list"
      >
        <ErrorItem
          className=""
          href="#example-error-1"
          id=""
          onClick={[Function]}
        >
          <li
            id=""
          >
            <a
              href="#example-error-1"
              id="null"
              onClick={[Function]}
            >
              Link to error with explanation
            </a>
          </li>
        </ErrorItem>
        <ErrorItem
          className=""
          href="#example-error-2"
          id=""
          onClick={[Function]}
        >
          <li
            id=""
          >
            <a
              href="#example-error-2"
              id="null"
              onClick={[Function]}
            >
              Link to error with explanation
            </a>
          </li>
        </ErrorItem>
      </ul>
    </div>
  </div>
</ErrorSummary>
`;

exports[`ErrorSummary doesn't render description or title when not provided 1`] = `
<ErrorSummary
  className=""
  description=""
  role="alert"
  tabIndex="-1"
>
  <div
    className="nhsuk-error-summary"
    role="alert"
    tabIndex="-1"
  >
    <h2
      className="nhsuk-error-summary__title"
    />
    <div
      className="nhsuk-error-summary__body"
    >
      <ul
        className="nhsuk-list nhsuk-error-summary__list"
      >
        <ErrorItem
          className=""
          href="#example-error-1"
          id=""
          onClick={[Function]}
        >
          <li
            id=""
          >
            <a
              href="#example-error-1"
              id="null"
              onClick={[Function]}
            >
              Link to error with explanation
            </a>
          </li>
        </ErrorItem>
        <ErrorItem
          className=""
          href="#example-error-2"
          id=""
          onClick={[Function]}
        >
          <li
            id=""
          >
            <a
              href="#example-error-2"
              id="null"
              onClick={[Function]}
            >
              Link to error with explanation
            </a>
          </li>
        </ErrorItem>
      </ul>
    </div>
  </div>
</ErrorSummary>
`;

exports[`ErrorSummary item matches snapshot 1`] = `
<li
  id=""
>
  <a
    href="#"
    id="null"
    onClick={[Function]}
  >
    Error item
  </a>
</li>
`;
